### ISSUE TITLE
Waku Relay scability in the Browser

### ISSUE THREAD

**Issue by fryorcraken** [status-im/js-waku#905]
## Problem

See https://forum.vac.dev/t/waku-v2-scalability-studies/142/9?u=fryorcraken for original post

tl;dr:

1. Browser nodes do not connect to each other with the currently available transport protocols (tcp, websocket).
2. Browser nodes do not advertise themselves via the currently available discovery protocols (static list, DNS Discovery, discv5).
3. If browser nodes were to advertise themselves, because of the ephemeral nature of a browser tab being opened, they would likely to poorly contribute to the Waku Relay network.

### 1. Enable browser to browser connection

- [ ] browser-to-browser connection using WebRTC:
  - [ ] https://github.com/status-im/js-waku/issues/20
  - [ ] https://github.com/status-im/js-waku/issues/681
- [ ] browser-to-browser connection using WebTransport?

Ref https://github.com/libp2p/specs/issues/220

### 2. Peer Exchange "Push"

Once browser-to-browser connection is possible, then browser need to advertise themselves in way that works for short live nodes.

- [ ] Peer exchange push RFC
- [ ] Implement it

### 3. Waku Browser Extension

- [ ] Scope and create a PoC that uses browser-to-browser connection
- [ ] Scope and create a PoC that uses a local/user control service node to access the Waku network

Notes:
- usage of the extension should be pseudo-transparent to the developer
- Waku apps with a browser such as Status could provide an interface equivalent to the extension

**Comment by weboko** [comment]
**Weekly update**:
_achieved_: draft of direct browser-browser RTC example https://github.com/waku-org/js-waku-examples/pull/260 
_next_: improve the example (connection re-usage), work on contentTopic based RTC example
_blocker_: none

**Comment by fryorcraken** [comment]
Need to review this:
1. new transport protocols (webrtc browser<>server, webtransport) are likely to be used for Waku scalability in general so that browser nodes can connect to status desktop nodes
2. relay in the browser is more for a privacy focused item, probably better handled in 2024.

For now, let's continue on the current work wich aims to chip a PoC that run Waku Relay over WebRTC to other node. Once done, we can review the work we want to do this year.

**Comment by weboko** [comment]
**Weekly update**:
- _achieved_: complete PoC using circuit relay
- _next_: research issue for next iteration and pause for now

**Comment by fryorcraken** [comment]
Icebox'd, see https://github.com/waku-org/pm/issues/50#issuecomment-1778451567
